### PR TITLE
Backport Copilot feature to `release-v2.8`

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description

- Backports Copilot feature to `release-v2.8`

### CODE REVIEW

File: libnavigation-core/api/current.txt [hunk 1]
```diff
@@ -678,11 +689,13 @@ package com.mapbox.navigation.core.routealternatives {
   }
 
   public final class AlternativeRouteMetadata {
+    method public int getAlternativeId();
```
Comment (dzinad) [inline libnavigation-core/api/current.txt:692]: Wondering where this came from. Is this Copilot related?

Comment (Guardiola31337) [reply libnavigation-core/api/current.txt:692]: This is definitely not coming from Copilot. Wondering if it's a merge conflict issue or a Metalava one. Will check. Thanks for flagging!

Comment (Guardiola31337) [reply libnavigation-core/api/current.txt:692]: I see that removing both entries and running `$> make core-check-api` is still ✅ 

Will remove them. I believe the issue is https://github.com/mapbox/mapbox-navigation-android/issues/6257

**Review by dzinad — APPROVED** [review]
